### PR TITLE
Fix `metric_custom_indicator`'s `good` and `total` metrics array

### PR DESCRIPTION
**Pull request by jloleysens**
Ensure that the all metrics specified in the array are applied to the SLO resource

Close https://github.com/elastic/terraform-provider-elasticstack/issues/544

### CODE REVIEW

**Comment by jloleysens** [comment]
Fix https://github.com/elastic/terraform-provider-elasticstack/issues/544

File: internal/kibana/slo.go [hunk 1]
```diff
@@ -541,6 +542,17 @@ func getSloFromResourceData(d *schema.ResourceData) (models.Slo, diag.Diagnostic
 		}
 
 	case "metric_custom_indicator":
+		goodMetricsRaw := d.Get(indicatorType + ".0.good.0.metrics").([]interface{})
```
Comment (cnasikas) [inline internal/kibana/slo.go:545]: It seems that in the SLO UI the `total` can also accept an array of metrics. Should we also support an array of metrics for it?

Screenshot 2024-01-29 at 11 11 02 AM

**Review by cnasikas — APPROVED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -99,6 +99,7 @@ jobs:
           - '8.8.2'
           - '8.9.2'
           - '8.10.3'
+          - '8.11.0'
```
Comment (jloleysens) [inline .github/workflows/test.yml]: Looks like something is not working when creating the custom metric SLO against this version of Kibana... Need to find the fix version, just testing out 8.11 for now.

Comment (jloleysens) [reply .github/workflows/test.yml]: This looks to be unrelated... Could use some help with the errors in tests though 😅

Comment (jloleysens) [reply .github/workflows/test.yml]: See 5ffaa40c401c9ef6d6dd7b1fbab59af6e3d8b765 🤦🏻

**Comment by tobio** [comment]
@wandergeek are you able to take a look over this one?

File: internal/kibana/slo.go [hunk 3]
```diff
@@ -541,34 +542,42 @@ func getSloFromResourceData(d *schema.ResourceData) (models.Slo, diag.Diagnostic
...
 		indicator = slo.SloResponseIndicator{
 			IndicatorPropertiesCustomMetric: &slo.IndicatorPropertiesCustomMetric{
 				Type: indicatorAddressToType[indicatorType],
 				Params: slo.IndicatorPropertiesCustomMetricParams{
 					Filter:         getOrNilString(indicatorType+".0.filter", d),
 					Index:          d.Get(indicatorType + ".0.index").(string),
 					TimestampField: d.Get(indicatorType + ".0.timestamp_field").(string),
-					Total: slo.IndicatorPropertiesCustomMetricParamsTotal{
-						Equation: d.Get(indicatorType + ".0.total.0.equation").(string),
-						Metrics: []slo.IndicatorPropertiesCustomMetricParamsTotalMetricsInner{ //are there actually instances where there are more than one 'good' / 'total'? Need to build array if so.
```
Comment (jloleysens) [inline internal/kibana/slo.go:553]: Yes there are 😄

**Review by wandergeek — APPROVED** [review]
Looks good to me, man! Thanks for the contribution.

**Pull request merged**